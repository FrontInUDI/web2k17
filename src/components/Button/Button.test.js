// import mountTemplate from '../../../test/mountTemplate'
import { shallow } from 'vue-test-utils'
import { createRenderer } from 'vue-server-renderer'
import UDIButton from './Button.vue'

/* eslint-disable no-undef */

test('should render a button', async () => {
  const renderer = createRenderer()
  const wrapper = shallow(UDIButton)

  expect(wrapper.text()).toContain('')
  expect(wrapper.hasClass('udi-button')).toBe(true)
  expect(wrapper.is('button')).toBe(true)
  renderer.renderToString(wrapper.vm, (_, str) => {
    expect(str).toMatchSnapshot()
  })
})

test('should render a link', async () => {
  const renderer = createRenderer()
  const wrapper = shallow(UDIButton, {
    attachToDocument: true,
    propsData: { link: true, href: 'http://localhost' }
  })

  expect(wrapper.text()).toContain('')
  expect(wrapper.hasClass('udi-link')).toBe(true)
  expect(wrapper.hasProp('link', true)).toBe(true)
  expect(wrapper.hasProp('href', 'http://localhost')).toBe(true)
  renderer.renderToString(wrapper.vm, (_, str) => {
    expect(str).toMatchSnapshot()
  })
})
