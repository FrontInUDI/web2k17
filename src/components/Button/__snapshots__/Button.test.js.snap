// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a button 1`] = `<button class="udi-button"></button>`;

exports[`should render a link 1`] = `<a href="http://localhost" class="udi-link"></a>`;
